### ISSUE TITLE
How to change the fontsize of pheatmap returned by plot_multiple_branches

### ISSUE THREAD

**Issue by ghost** [cole-trapnell-lab/monocle-release#183]
I am using the plot_multiple_branches function and was wondering how to change the font sizes of the returned heatmap - in particular, making the gene row names smaller so they do not over lap. I have set return_heatmap = TRUE so I have the pheatmap saved as a variable, but I do not know where to go from there.  If I make the figure larger so the gene names do not overlap, the legend text will be too small. Thanks for your help!

Here is an example of code I am using from the Olsson dataset analysis tutorial 
`plot_multiple_branches_heatmap(URMM_all_fig1b[unique(c(positive_score_genes, negtive_score_genes)),],
                                branches=c(3, 4, 5),
                                branches_name=c("Granulocyte", "Monocyte", "Ery/Meg"),
                                show_rownames=T,
                                num_clusters=4)`

**Comment by atchen** [comment]
If there isn't a way to explicitly control the font size, I do realize I can choose to show fewer genes based on their qval.

`sig_genes <- row.names(subset(diff_test_res, qval < 1e-60))
plot_multiple_branches_heatmap(celldataset[sig_genes,],
                               branches=c(1),
                               branches_name=c('state1'),
                               show_rownames=T,
                               num_clusters=3, return_heatmap = TRUE) `

**Comment by Xiaojieqiu** [comment]
hey @atchen , unfortunately we have not export the font_size argument in the `plot_multiple_branches_heatmap` function yet. But I will export it for our next release. In the meanwhile, you can either chose fewer genes, increase the vertical length of the heatmap or set `return_heatmap` equal to `TRUE` to return the object for creating the heatmap. 

```
ph_res <- plot_multiple_branches_heatmap(URMM_all_fig1b[unique(c(positive_score_genes, negtive_score_genes)),], branches=c(3, 4, 5), branches_name=c("Granulocyte", "Monocyte", "Ery/Meg"), show_rownames=T, num_clusters=4, return_heatmap = TRUE)
ph_res$gtable$vp$gp$fontsize <- 5 # reset the font size (I have not test this so it may not work) 
grid::grid.rect(gp=grid::gpar("fill", col=NA))
grid::grid.draw(tmp$gtable)
```

**Comment by ShaowenJ** [comment]
Yes, it does not work to change the fontsize. Any one knows a better method?

**Comment by Weiyifan** [comment]
> hey @atchen , unfortunately we have not exported the font_size argument in the `plot_multiple_branches_heatmap` function yet. But I will export it for our next release. In the meanwhile, you can either chose fewer genes, increase the vertical length of the heatmap or set `return_heatmap` equal to `TRUE` to return the heatmap object and recreate the heatmap with small font size .
> 
> ```
> ph_res <- plot_multiple_branches_heatmap(URMM_all_fig1b[unique(c(positive_score_genes, negtive_score_genes)),], branches=c(3, 4, 5), branches_name=c("Granulocyte", "Monocyte", "Ery/Meg"), show_rownames=T, num_clusters=4, return_heatmap = TRUE)
> ph_res$gtable$vp$gp$fontsize <- 5 # reset the font size (I have not test this so it may not work) 
> grid::grid.rect(gp=grid::gpar("fill", col=NA))
> grid::grid.draw(ph_res$gtable)
> ```

Does not work any more

**Comment by zhaoshuoxp** [comment]
Here is one solution: @Weiyifan @ShaowenJ 
Download source code of monocle, edit R/plotting.R on line 1691, add a fontsize parameter (i.e font_size=10) to the function plot_genes_branched_heatmap(), then quote your parameter on line1864, i.e. edit fontsize = 6 to fontsize = font_size.
load your edited local monocle package by devtools::load_all("/path/to/edited/monocle/")

Then you should be able to use font_size in the plot_genes_branched_heatmap function.

Quanyi
